shapeshift/web
# feat: remove foxy rebase history

## Description

This PR:

- removes rebase functionality from investor-foxy and web consumption
- removes the notion of rebase from web `useBalanceChartData` and `txHistorySlice`
- removes now unused Infura env var

## Pull Request Type


## Issue (if applicable)



closes https://github.com/shapeshift/web/issues/5736

## Risk



- EVM accounts (and consequently, net worth) balances may be borked, test accordingly

## Testing



- Ensure FOXy price is still happy 
- Ensure FOXy DeFi opportunity, user data (balances) and functionality (deposit/withdraw/claim) is still happy
- Ensure EVM accounts' balances looks sane against prod
- Ensure net worth looks sane against prod

### Engineering



- ☝🏽 
- grep for rebase and ensure all runtime occurences are gone

### Operations



- ☝🏽 

## Screenshots (if applicable)

### FOXy balance

- Prod

<img width="1023" alt="Screenshot 2023-12-05 at 11 35 19" src="https://github.com/shapeshift/web/assets/17035424/9b7291f3-02c1-47f6-9680-15fb83dd2b9a">

- This diff

<img width="967" alt="Screenshot 2023-12-05 at 11 35 27" src="https://github.com/shapeshift/web/assets/17035424/c590ccbf-f4b3-4a6e-8d95-bb82ae54b0f8">

### Net worth

- Prod

<img width="983" alt="Screenshot 2023-12-05 at 11 34 54" src="https://github.com/shapeshift/web/assets/17035424/0db4f5b7-3704-4df1-8560-9efe3f29347a">

### FOXy DeFi

- This diff

<img width="764" alt="image" src="https://github.com/shapeshift/web/assets/17035424/f5bea66b-8e5e-4bbd-aaeb-ce7e3b520702">

- Prod

<img width="791" alt="image" src="https://github.com/shapeshift/web/assets/17035424/1b516f58-f039-4f7f-934f-3d0c0b1e2d1a">

- This diff

 
<img width="991" alt="Screenshot 2023-12-05 at 11 35 00" src="https://github.com/shapeshift/web/assets/17035424/b5dde741-b4e2-4f18-9147-678834004fe9">

### Perf

- Prod

<img width="970" alt="Screenshot 2023-12-05 at 11 46 30" src="https://github.com/shapeshift/web/assets/17035424/3250aa87-dd2b-451e-a6b6-d6251ba66cc8">
<img width="975" alt="Screenshot 2023-12-05 at 11 46 23" src="https://github.com/shapeshift/web/assets/17035424/7933e826-2927-46ce-bf40-aac208a129b0">

Notes:

1.  in prod, Infura and *our* JSON-RPC node requests are decoupled. That's because develop isn't whitelisted for Infura, so we use our node as a bandaid currently, which makes FOxy unreliable in localhost currently
2. JSON-RPC requests are batched, and prod being faster, the batching may be more efficient, so the number of requests in prod vs. here for the same account isn't really indicative of anything

- This diff

<img width="994" alt="Screenshot 2023-12-05 at 11 44 46" src="https://github.com/shapeshift/web/assets/17035424/2698dc9a-a84a-4f63-8a3a-5e16fb4c997d">
